cpp/include/cuspatial/trajectory.hpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/cudf.h>
+
+namespace cuspatial {
+
+/**
+ * @Brief deriving trajectories from points (x/y relative to an origin), timestamps and objectids
+ * by first sorting based on id and timestamp and then group by id.
+
+ * @param[in/out] coor_x:  x coordiantes reative to a camera origin (before/after sorting)
+ * @param[in/out] coor_y:  y coordiantes reative to a camera origin (before/after sorting)
+ * @param[in/out] oid: object (e.g., vehicle) id column (before/after sorting); upon completion, unqiue ids become trajectory ids
+ * @param[in/out] ts: timestamp column (before/after sorting)
+ * @param[out] tid: trajectory id column (see comments on oid)
+ * @param[out] len: #of points in the derived trajectories
+ * @param[out] pos: position offsets of trajectories used to index coor_x/coor_y/oid/ts after completion
+
+ * @returns the number of derived trajectory
+ */
+int coord_to_traj(gdf_column& coor_x,gdf_column& coor_y,gdf_column& oid, gdf_column& ts,
```

Comment:
I believe that the three output columns have equal length. Correct? If so, return a table that contains the three output columns. Much cleaner.

```suggestion
cudf::table coord_to_traj(const gdf_column& coords_x, const gdf_column& coords_y,
                          const gdf_column& object_ids, const gdf_column& timestamps);
```